Automattic/jetpack
# Resolve Sync Errors from empty edge case and WP.com returning concurrent_request_error

Fixes #16718 

#### Changes proposed in this Pull Request:
* Adds appropriate sanitization to ensure that errors are not triggered when items_ids is empty or WP.com responds with an error when Syncing content.

#### Jetpack product discussion
* Bug Fix

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
* fetch_items_by_id is protected so testing needs to be performed on peek_by_id
* Execute `wp shell`
* $queue = new Automattic\Jetpack\Sync\Queue( 'blue' );
* $queue->peek_by_id( null );
* Verify an empty array is returned. This should occur for all inputs except an array with valid sync ids.
* To get valid sync ids look at the options table for options with a name like jpsq_blue-%  (name of queue instantiated earlier)

Unit Tests have been added for peek_by_id for various input types and values.

#### Proposed changelog entry for your changes:
* Resolve errors triggered from not properly sanitizing/verifying inputs during Sync process. Identified by @HFigarella, Thank you.
